src/rt/pal/pal_linux.h

```diff
@@ -0,0 +1,398 @@
+#pragma once
+
```

Comment:
This file is interleaving a lot of Linux-specific and generic-POSIX code, which will make bringing up the next platform painful.  The `sandbox_pal` branch includes [platform abstractions that hide the minimal bits that we need to paper over the differences between platforms](https://github.com/microsoft/verona/tree/sandbox_pal/experiments/process_sandbox/platform).  In particular, take a look at the [poller_epoll.h](https://github.com/microsoft/verona/blob/sandbox_pal/experiments/process_sandbox/platform/poller_epoll.h) and [poller_kqueue.h](https://github.com/microsoft/verona/blob/sandbox_pal/experiments/process_sandbox/platform/poller_kqueue.h) files.  These are very close to what you want and could easily be made more general.